controllers/discovery/searchDomain.js

```diff
@@ -11,7 +11,7 @@ const {sequelize} = require('../../databases/models');
 const SearchDomain = async (req, res) => {
   const {q} = req.query;
   const {userId} = req;
-  if (q.length < 2)
+  if (!q || q?.length < 2)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code has improved error handling by checking if the query parameter `q` is not null before checking its length. This prevents a potential TypeError in case `q` is undefined.
```diff
-  if (q.length < 2)
+  if (!q || q?.length < 2)
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->